oecophylla/distance/mash.rule

```diff
@@ -12,137 +12,26 @@ rule mash_sketch:
         forward = qc_dir + "{sample}/filtered/{sample}.R1.trimmed.filtered.fastq.gz",
         reverse = qc_dir + "{sample}/filtered/{sample}.R2.trimmed.filtered.fastq.gz"
     output:
-        sketch = distance_dir + '{sample}/mash/{sample}.msh'
+        distance_dir + '{sample}/mash/{sample}.msh'
     params:
-        mash = config['software']['mash'],
-        seqtk = config['software']['seqtk'],
-        mash_params = config['params']['mash']['other'],
-        depth = config['params']['mash']['depth']
+        env = config['envs']['distance'],
+        # divided by 2 because both forward and reverse contribute to the sketch
+        size = int(config['params']['mash']['sketch_size']) / 2
     threads:
         1
     log:
         distance_dir + "logs/mash_sketch.sample=[{sample}].log"
     benchmark:
-        "benchmarks/distance/mash_sketch.sample=[{sample}].json"
-    run:
-        output_base = os.path.splitext(output['sketch'])[0]
+        "benchmarks/distance/mash_sketch.sample=[{sample}].txt"
+    shell:
+        # -r is for reads instead of genomes for mash sketch;
+        # remove suffix because mash will automatically add .msh
+        '''set +u; {params.env}; set -u
+        output_basename=`basename {output[0]} .msh`
```

Comment:
why `output[0]`, is that always the first sample?

Reply:
yes. actually only using forward reads.

Reply:
note it's for output - there is only one output file needed and specified.